Assets/Mirror/Runtime/SyncList.cs

```diff
@@ -292,6 +292,17 @@ public int FindIndex(Predicate<T> match)
             return -1;
         }
 
+        public T Find(Predicate<T> match) => objects[FindIndex(match)];
```

Comment:
Ideally this would behave exactly like List<T>.Find,  otherwise this will be very confusing to users.

If there is no match your method throws IndexOutOfRangeException
List<T>.Find will not throw exception, but return default according to the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.find?view=netframework-4.8).

Reply:
SyncList.Find now returns default on no match.
Unit tests have been added.